### PR TITLE
[TE] frontend - aaronucsd/added new duration custom template for the …

### PR DESCRIPTION
**Pull request by aaronucsd**
screen shot 2018-03-08 at 1 16 43 pm

…dashboard app table

1. Added a new custom template for Start/Duration
2. updated mock data for dashboard table
3. Created new util for anomaly and included test
4. Updated existing util to use new anomaly util
5. ran all test suite for sanity check
6. Added in eslint scripts in package.json

### CODE REVIEW

File: thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js [hunk 1]
```diff
@@ -311,7 +311,7 @@ export default Route.extend({
       alertDimension: exploreDimensions,
       isReplayPending: isPresent(jobId) && jobId !== -1,
       alertHasDimensions: isPresent(exploreDimensions),
-      timeRangeOptions: setUpTimeRangeOptions(['3m'], duration),
+      timeRangeOptions: setUpTimeRangeOptions(['3m', '1m'], duration),
```
[the remark this answers is not in the document]
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js]: Good catch. It's not. I made a change to test. Will revert.

File: thirdeye/thirdeye-frontend/app/pods/custom/anomalies-table/start-duration/template.hbs [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {{!-- TODO: Replace css classes below with te-anomaly-table styles --}}
-<p class="anomalies-table__start">
+{{#link-to 'rootcause' (query-params anomalyId=record.id)
+  class="te-anomaly-table__link"}}
   {{record.start}}
-</p>
+{{/link-to}}
 <p class="anomalies-table__duration">
   {{record.duration}}
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/pods/custom/anomalies-table/start-duration/template.hbs]: I think we can remove the `TODO` comment and the classname on line 6. I had it there, thinking that I was going to do something about it later, but it doesn't seem like it needs any additional styles to apply here.

File: thirdeye/thirdeye-frontend/app/pods/home/route.js [hunk 3]
```diff
@@ -61,6 +62,8 @@ export default Route.extend({
       const changeFloat = (current - baseline) / baseline;
       anomaly.change = floatToPercent(changeFloat);
       anomaly.humanizedChange = humanizeChange(changeFloat);
+      anomaly.duration = getFormatedDuration(anomaly.start, anomaly.end);
+      anomaly.start = moment(anomaly.start).format('MMM D, hh:mm A');
```
[the remark this answers is not in the document]
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/pods/home/route.js]: Good to know. WIll use this in the template and drop line 66. Thanks!

File: thirdeye/thirdeye-frontend/app/utils/anomaly.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import moment from 'moment';
+import { isPresent } from "@ember/utils";
+import _ from 'lodash';
+
+/**
+ * Formats anomaly duration property for display on the table
+ * @param {Number} anomalyStart - the anomaly start time
+ * @param {Number} anomalyEnd - the anomaly end time
+ * @returns {String} the fomatted duration time
+ * @example anomaly.duration = getFormatedDuration(anomaly.start, anomaly.end);
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/app/utils/anomaly.js]: Can we elaborate on the example like the following?
```
@example getFormatDuration(1234567, 456789) = 'April 5th, 2018' // I made this up. It's not accurate
```

File: thirdeye/thirdeye-frontend/app/utils/anomaly.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ */
+export function getFormatedDuration(anomalyStart, anomalyEnd) {
+  const startMoment = moment(anomalyStart);
+  const endMoment = moment(anomalyEnd);
+  const anomalyDuration = moment.duration(endMoment.diff(startMoment));
+  const days = anomalyDuration.get("days");
```
[the remark this answers is not in the document]
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/utils/anomaly.js]: Ah right. It was copied from existing codes. Wonder how those got by eslint then. Good catch.

File: thirdeye/thirdeye-frontend/package.json [hunk 6]
```diff
@@ -12,7 +12,10 @@
   "scripts": {
     "build": "ember build",
     "start": "ember server --proxy http://localhost:1426",
-    "test": "ember test"
+    "test": "ember test",
+    "eslint": "node_modules/.bin/eslint",
+    "eslint-app": "npm run eslint app/",
+    "eslint-tests": "npm run eslint tests/"
```
Comment (ttbach) [inline thirdeye/thirdeye-frontend/package.json]: woohoo! 🎉

File: thirdeye/thirdeye-frontend/app/mirage/fixtures/applicationAnomalies.js [hunk 7]
```diff
@@ -3,8 +3,8 @@
 export default [
   {
     id: 1,
-    start: 12345,
-    end: 23456,
+    start: 1491804013000,
```
[the remark this answers is not in the document]
Comment (aaronucsd) [reply thirdeye/thirdeye-frontend/app/mirage/fixtures/applicationAnomalies.js]: Neat. Will update to use. Thanks!

**Pull request merged**